contracts/debt/IncurDebtV1.sol

```diff
@@ -0,0 +1,458 @@
+
+        if (depositedCollateralAfterRepay > 0) {
+            if (_tokenToReceiveExcess == sOHM) {
+                IERC20(sOHM).safeTransfer(msg.sender, depositedCollateralAfterRepay);
+            } else if (_tokenToReceiveExcess == gOHM) {
+                IERC20(sOHM).approve(staking, depositedCollateralAfterRepay);
+                IStaking(staking).wrap(msg.sender, depositedCollateralAfterRepay);
+            }
+        }
+
+        emit DebtPaidWithCollateralAndWithdrawTheRest(
+            msg.sender,
+            paidDebt,
+            borrowers[msg.sender].collateralInSOHM,
+            borrowers[msg.sender].debt,
+            totalOutstandingGlobalDebt,
+            depositedCollateralAfterRepay
+        );
+    }
+
+    /**
+     * @notice deposits OHM to pay debt
+     * - msg.sender must be a borrower
+     * - msg.sender's OHM allowance must be >= _ohmAmount
+     * - borrower must have outstanding debt
+     * @param _ohmAmount amount of OHM to borrow
+     */
+    function repayDebtWithOHM(uint256 _ohmAmount) external override isBorrower(msg.sender) {
+        Borrower storage borrower = borrowers[msg.sender];
+
+        if (borrower.debt == 0) revert IncurDebtV1_BorrowerHasNoOutstandingDebt(msg.sender);
+        IERC20(OHM).safeTransferFrom(msg.sender, address(this), _ohmAmount);
+
+        totalOutstandingGlobalDebt -= _ohmAmount;
+        borrower.debt = uint128(borrower.debt - _ohmAmount);
+
+        IERC20(OHM).approve(treasury, _ohmAmount);
+        ITreasury(treasury).repayDebtWithOHM(_ohmAmount);
+
+        emit DebtPaidWithOHM(msg.sender, _ohmAmount, borrower.debt, totalOutstandingGlobalDebt);
+    }
+
+    /**
+     * @notice repays debt using collateral and returns remaining tokens to borrower
+     * - onlyOwner (or governance)
+     * - sends remaining tokens to owner in sOHM
+     * @param _borrower the address that will interact with contract
+     * @param _to where to send remaining sOHM
+     */
+    function forceRepay(address _borrower, address _to) external override onlyGovernor {
+        (uint256 collateralAfterDebtPayment, uint256 paidDebt) = _repay(_borrower);
+
+        assignBorrowerInfoToZero(_borrower);
+        if (collateralAfterDebtPayment > 0) IERC20(sOHM).safeTransfer(_to, collateralAfterDebtPayment);
+
+        emit ForceDebtPayWithCollateralAndWithdrawTheRest(
+            _borrower,
+            paidDebt,
+            borrowers[msg.sender].collateralInSOHM,
+            borrowers[msg.sender].debt,
+            totalOutstandingGlobalDebt,
+            collateralAfterDebtPayment
+        );
+    }
+
+    /**
+     * @notice seize and burn _borrowers collateral and forgive debt
+     * - will burn all collateral, including excess of debt
+     * - onlyGovernance
+     * @param _borrower the account to seize
+     */
+    function seize(address _borrower) external override onlyGovernor isBorrower(_borrower) {
+        uint256 paidDebt = _repayDebtWithCollateralByGovernor(_borrower);
```

Comment:
Should be able to just have same logic as `forceRepay()`. Then just burn, shouldn't need `_repayDebtWithCollateralByGovernor()` since still just using `_repay()`